### PR TITLE
support cargo's --quiet arg

### LINKED ISSUE #1: FR: add `--quiet` flag to shebang
this makes the output nicer, in my opinion:
```
warning: `package.edition` is unspecified, defaulting to `2021`
   Compiling hello_world v0.0.0 (/SNIP/cargo-new-script)
    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
     Running `/SNIP/debug/hello_world`
Hello, world!
```
vs
```
Hello, world!
```

### PR DESCRIPTION
**Pull request by pickx**
fixes #1 
this is a trivial change, but implementation suggestions are welcome. 
thanks.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -36,12 +36,15 @@ fn main() -> anyhow::Result<()> {
...
-        "#!/usr/bin/env -S cargo +nightly -Zscript"
+fn shebang(nightly: bool, quiet: bool) -> String {
+    let cargo_invocation = if nightly {
+        "-S cargo +nightly -Zscript"
     } else {
-        "#!/usr/bin/env cargo"
-    }
+        "cargo"
+    };
+    let quiet_arg = quiet.then_some(" --quiet").unwrap_or("");
```
Comment (avsaase) [inline src/main.rs]:
```suggestion
    let quiet_arg = if quiet { " --quiet" } else { "" };
```

File: src/main.rs [hunk 2]
```diff
@@ -81,6 +84,9 @@ struct NewSciptArgs {
     /// Create a shebang line that uses the stable toolchain. Currently, this does not generate a runnable script because `cargo script` requires nightly.
     #[arg(long)]
     stable: bool,
+
```
Comment (avsaase) [inline src/main.rs:87]: Please add a doc comment to explain what the new flag does.

**Review by avsaase — CHANGES_REQUESTED** [review]
Great idea! I think this should be the default so maybe you can invert the flag and rename it to `verbose-script` (to indicate that it's about the generated script and not the tool)?

**Comment by pickx** [comment]
> I think this should be the default
I agree. thanks for the quick review.

**Review by avsaase — APPROVED** [review]
Thanks for the contribution! Let's hope this lands on stable soon and we'll get `cargo new` support so this will no longer be needed.

**Pull request merged**